CRAMv3.tex

```diff
@@ -166,26 +166,29 @@ \subsubsection*{Example of writing to bit stream}
 \texttt{> echo "obase=2; ibase=16; B070" \textbar{} bc\\
 1011000001110000}
 
-When reading the bits from the bit sequence it must be known that only 12 bits 
-are meaningful and the bit stream should not be read after that. 
+When reading the bits from the bit sequence, only the first 12 bits are meaningful and the remaining 4 will should be discarded.
 
-\subsubsection*{Note on writing to bit stream}
+\subsubsection*{Note on reading from and writing to bit stream}
 
-When writing to a bit stream both the value and the number of bits in the value 
-must be known. This is because programming languages normally operate with bytes 
-(8 bits) and to specify which bits are to be written requires a bit-holder, for 
-example an integer, and the number of bits in it. Equally, when reading a value 
-from a bit stream the number of bits must be known in advance. In case of prefix 
-codes (e.g. Huffman) all possible bit combinations are either known in advance 
-or it is possible to calculate how many bits will follow based on the first few 
-bits. Alternatively, two codes can be combined, where the first contains the number 
-of bits to read. 
+When reading and writing to a bit stream our numeric values are
+typically held in a byte oriented data type, such as an 8-bit or
+32-bit integer.
+The bit stream itself does not explicitly store the number of bits
+per value, and it will vary by context, so we must know this by other means.
```

Comment:
This answers #812.

The examples that follow are useful but would be better defined with their respective codecs. E.g., § 13.5 "[Beta encoding: codec ID 6](https://samtools.github.io/hts-specs/CRAMv3.pdf#subsection.13.5)" should define the resulting value from the bit stream is a 32-bit integer, meaning the _length_ value is limited to 32 (or 31 if the value is always supposed be non-negative?). This is the context required to meaningfully read from the bit stream.